File: Alphabet-Spam/alphabetspam.cpp
#include <iostream>

using namespace std;

int main()
{
   string s;
   cin >> s;

   double ws, lc, uc, sy;

   for (int i = 0; i < s.length(); i++)
   {
       if (islower(s[i]))
          lc++;
       else if (isupper(s[i]))
          uc++;
       else if (s[i] == '_')
          ws++;
       else
          sy++;
   }

   double wsAns = ws / s.length();
   double lcAns = lc / s.length();
   double ucAns = uc / s.length();
   double sAns = sy / s.length();

   cout << wsAns << endl << lcAns << endl << ucAns << endl << sAns << endl; 


   return 0;
}

File: solutions/alphabetspam.py
ws, l, u, s = 0, 0, 0, 0
for c in (x := input()):
    if c == '_':
        ws += 1
    elif c.islower():
        l += 1
    elif c.isupper():
        u += 1
    else:
        s += 1

print(f"{ws/len(x):.15f}")
print(f"{l/len(x):.15f}")
print(f"{u/len(x):.15f}")
print(f"{s/len(x):.15f}")
